src/components/cart-row/cart-row.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState } from "react";
+import { Link } from "react-router-dom";
+import { CartItem } from "../../entities/cart-item";
+import CartService from "../../services/cart-service";
+
+interface Props {
+  cartItem: CartItem;
+  cart: CartItem[];
+  setCart: (newCart: CartItem[]) => void;
+}
+
+const CartRow = ({ cartItem, cart, setCart }: Props) => {
+  const { product, quantity } = cartItem;
+  const [newQuantity, setQuantity] = useState(quantity);
+
+  const handleQuantityChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newQuantity = parseInt(event.target.value, 10);
```

Comment:
You can simply write const _newQuantity = parseInt(event.target.value)_ without radix, as it is an optional prop.